public/scripts/controllers/BooksShowController.js

```diff
@@ -0,0 +1,49 @@
+angular
+  .module("angularBooks")
+  .controller("BooksShowController", BooksShowController);
+
+BooksShowController.$inject = ["$http", "$routeParams", "$location"];
+
+function BooksShowController(   $http,   $routeParams, $location ) {
+  var vm = this;
+
+  $http({
+    method: "GET",
+    url: "https://super-crud.herokuapp.com/books/" + $routeParams.id
+  }).then(function successCallback(json) {
+    vm.book = json.data;
+  }); // end of GET request
+
+
+  vm.editBook = function(book) {
+    $http({
+      method: "PUT",
+      url: "https://super-crud.herokuapp.com/books/" + book._id,
+      data: {
```

Comment:
I'm pretty sure you can just set it as data: book (since the function takes the book as an argument), it's not wrong, just redundant